File: mediaeffectswithopengles/src/main/java/com/dastanapps/mediaeffectswithopengles/MainActivity.java
package com.dastanapps.mediaeffectswithopengles;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Toast;

import com.dastanapps.mediaeffectswithopengles.databinding.ActivityMainBinding;
import com.dastanapps.mediaeffectswithopengles.record.GLSurfaceViewPlayerActivity;
import com.dastanapps.mediaeffectswithopengles.record.RecordGLwithoutRenderOnScreen;
import com.dastanapps.mediaeffectswithopengles.record.RecordSurfaceActivity;

public class MainActivity extends AppCompatActivity implements AdapterView.OnItemClickListener {

    ActivityMainBinding activityMainBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        ArrayAdapter<String> menuAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1,
                getResources().getStringArray(R.array.menu_opts));
        activityMainBinding.list.setAdapter(menuAdapter);
        activityMainBinding.list.setOnItemClickListener(this);

    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        switch (position) {
            case 0:
                RecordGLwithoutRenderOnScreen recordGLwithoutRenderOnScreen = new RecordGLwithoutRenderOnScreen();
                recordGLwithoutRenderOnScreen.recordOpenGLVideo();
                Toast.makeText(this, "Search test.currentDate.mp4 file in File Storage", Toast.LENGTH_SHORT).show();
                break;
            case 1:
                startActivity(RecordSurfaceActivity.class);
                break;
            case 2:
                startActivity(GLSurfaceViewPlayerActivity.class);
                break;
        }
    }

    private void startActivity(Class clz) {
        startActivity(new Intent(this, clz));
    }
}

File: players/src/main/java/com/dastanapps/players/MainActivity.kt
package com.dastanapps.players

import android.content.Intent
import android.os.Bundle
import android.widget.AdapterView
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    lateinit var exoPlayerHelper: ExoPlayerHelper
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        exoPlayerHelper = ExoPlayerHelper(this)
        val listView = findViewById<ListView>(R.id.listview)
        listView.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            when (position) {
                0 -> startActivity(Intent(this, ExoPlayerActivity::class.java))
                1 -> startActivity(Intent(this, ExoPlayer242::class.java))
                2 -> startActivity(Intent(this, MediaPlayerActivity::class.java))
                3 -> startActivity(Intent(this, TextureViewActivity::class.java))
            }
        }
    }

    override fun onStart() {
        super.onStart()
        exoPlayerHelper.onStart()
    }

    override fun onResume() {
        super.onResume()
        exoPlayerHelper.onStart()
        exoPlayerHelper.playUrl("http://www.panacherock.com/downloads/mp3/01_Sayso.mp3",object : ExoPlayerHelper.Listener() {

        })
    }

    override fun onPause() {
        super.onPause()
        exoPlayerHelper.onStop()
    }

    override fun onStop() {
        super.onStop()
        exoPlayerHelper.onStop()
    }
}
